File: 169-MajorityElement/169-MajorityElement.js
// Last updated: 10/03/2026, 10:03:34
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let n = nums.length;
    for (let i = 0; i < n; i++) {
    let count = 0;
        for (let j = 0; j < n; j++) {
            if (nums[i] == nums[j]) {
                count++;
            }
        }
        if (count > n / 2) {
            return nums[i];
        }
    }


};

File: 169-MajorityElement/169-MajorityElement.cpp
// Last updated: 13/05/2026, 13:10:51
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (int num : nums) {
            mp[num]++;

            if (mp[num] > nums.size() / 2) {
                return num;
            }
        }
    return -1;
    }
};





// class Solution {
// public:
//     int majorityElement(vector<int>& nums) {
//         int count = 0;
//         int candidate = 0;

//         for(int num : nums) {
//             if(count == 0) {
//                 candidate = num;
//             }

//             if(num == candidate) {
//                 count++;
//             } else {
//                 count--;
//             }
//         }

//         return candidate;
//     }
// };